drivers/include/periph/gpio.h

```diff
@@ -151,109 +61,103 @@ typedef enum {
 } gpio_flank_t;
 
 /**
- * @brief Signature for function called from interrupt context after an external event
- *
- * @param[in] arg       optional context for the callback
- */
-typedef void (*gpio_cb_t)(void *arg);
-
-/**
- * @brief Initialize GPIO pin as output pin
+ * @brief   Initialize the given pin as general purpose input or output
  *
- * @param[in] dev       the GPIO pin to set as output
- * @param[in] pullup    define if pull resistors should be used
+ * @param[in] pin       pin to initialize
+ * @param[in] dir       direction for the pin, input or output
+ * @param[in] pullup    define what pull resistors should be used
  *
  * @return              0 on success
- * @return              -1 on pull-up mode unavailable
+ * @return              -1 on error
  */
-int gpio_init_out(gpio_t dev, gpio_pp_t pullup);
+int gpio_init(gpio_t pin, gpio_dir_t dir, gpio_pp_t pullup);
 
 /**
- * @brief Initialize a GPIO pin in input mode
+ * @brief   Initialize a GPIO pin for external interrupt usage
  *
- * @param[in] dev       the GPIO pin to use as input
- * @param[in] pullup    define if pull resistors should be used
+ * The registered callback function will be called in interrupt context every
+ * time the defined flank(s) are detected.
+ *
+ * The interrupt is activated automatically after the initialization.
+ *
+ * @param[in] pin       pin to initialize
+ * @param[in] pullup    define what pull resistors should be enabled
+ * @param[in] flank     define the active flank(s)
+ * @param[in] cb        callback that is called from interrupt context
+ * @param[in] arg       optional argument passed to the callback
  *
  * @return              0 on success
- * @return              -1 on pull-up mode unavailable
+ * @return              -1 on error
  */
-int gpio_init_in(gpio_t dev, gpio_pp_t pullup);
+int gpio_init_exti(gpio_t pin, gpio_pp_t pullup, gpio_flank_t flank,
```

Comment:
In this case I'd suggest to stay with the current name. `..._isr` brings the same problem like `..._exti` as you described above. `.._callback` appears too specific to me.